ui/src/FindBar.tsx

```diff
@@ -1,23 +1,53 @@
 import React from "react";
+import { convertToAnnotationId } from './selectors/annotation'
 import { EventBus } from "./types/pdfjs-viewer";
+import { ScholarReaderContext } from "./state";
+import { BoundingBox } from "./types/api";
+
+interface FindBarProps {
+  mappingToBounds: Map<String, BoundingBox>;
+  matches: string[];
+  selectedSymbol: string;
+}
 
 interface FindBarState {
   currentMatch: number | null;
   matchCount: number | null;
+  mode: string,
   /**
    * Event bus for the pdf.js application. The logic for performing text search within the PDF is
    * provided by pdf.js. This event bus is needed for communicating with the pdf.js code.
    */
   pdfJsEventBus?: EventBus;
 }
 
-export class FindBar extends React.PureComponent<{}, FindBarState> {
-  constructor(props: {}) {
-    super(props);
-    this.state = {
-      currentMatch: null,
-      matchCount: null,
-    };
+const initialState = {
+  currentMatch: null,
+  matchCount: null,
+  mode: 'hidden',
+}
+
+export class FindBar extends React.PureComponent<FindBarProps, FindBarState> {
+  state = initialState;
+  static contextType = ScholarReaderContext;
+  context!: React.ContextType<typeof ScholarReaderContext>;
+
+  /**
+   * This is considered an anti pattern in React (https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html),
+   * unfortunately since we are using this component for both string search (which does update match count without props) and symbol search
+   * we have to derive the match count and current match from the props. In the future we should probably divide these
+   * into separate components.
```

Comment:
I'm interested in your commend "divide these into separate components". I wonder if we can have a quick call about what a better way to organize this code would be. I'm planning on cleaning up the connection to the Mozilla code, and I might be able to implement a cleaner version of this as part of those efforts.